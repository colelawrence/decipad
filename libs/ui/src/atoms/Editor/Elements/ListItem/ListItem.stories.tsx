import { Meta, Story } from '@storybook/react';
import { ListItemElement } from './ListItem';

export default {
  title: 'Atoms/Editor/Elements/Title',
  component: ListItemElement,
  args: {
    children: 'Notebook Title',
  },
} as Meta;

interface ArgTypes {
  children: string;
}

export const Title: Story<ArgTypes> = (args) => (
  <ListItemElement
    attributes={{ 'data-slate-leaf': true, 'data-slate-node': 'element' }}
    leaf={{ text: '' }}
    text={{ text: '' }}
    nodeProps={{ styles: { root: { css: null } } }}
  >
    {args.children}
  </ListItemElement>
);
