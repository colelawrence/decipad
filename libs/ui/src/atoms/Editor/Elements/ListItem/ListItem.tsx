import { PlatePluginComponent } from '@udecode/plate';

export const ListItemElement: PlatePluginComponent = ({
  children,
  attributes,
}) => {
  return <li {...attributes}>{children}</li>;
};
