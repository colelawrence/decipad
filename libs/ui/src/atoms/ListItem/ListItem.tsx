import { css } from '@emotion/react';
import { FC, ReactNode } from 'react';
import { p16Regular } from '../../primitives';

const styles = css(p16Regular);

type ListItemProps = {
  readonly children?: ReactNode;
};
export const ListItem = ({ children }: ListItemProps): ReturnType<FC> => {
  return <div css={styles}>{children}</div>;
};
