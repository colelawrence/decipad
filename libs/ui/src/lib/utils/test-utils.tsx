import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

import { Type } from '@decipad/language';
import { theme, ResultsContextProvider } from '../..';

const results = {
  test: {
    type: Type.build({
      type: 'number',
      unit: {
        unit: 'apples',
      },
    } as any),
    value: [20],
    errors: [],
  },
  noResult: null,
};

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <ResultsContextProvider value={results}>
        {children}
      </ResultsContextProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
