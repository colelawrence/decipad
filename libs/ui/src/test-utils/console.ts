export const mockConsoleWarn = () => {
  const originalConsoleWarn = console.warn;
  const mockConsoleWarn: jest.MockedFunction<typeof console.warn> = jest.fn();

  beforeEach(() => {
    console.warn = jest.fn();
    mockConsoleWarn.mockClear();
  });
  afterEach(() => {
    console.warn = originalConsoleWarn;
  });

  return mockConsoleWarn;
};
