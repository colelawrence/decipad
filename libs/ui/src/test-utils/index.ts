export * from './menu';
export * from './language';
