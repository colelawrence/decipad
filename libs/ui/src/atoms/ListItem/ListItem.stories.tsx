import { Meta, Story } from '@storybook/react';
import { ListItem } from './ListItem';

const args = {
  children: 'Item',
};

export default {
  title: 'Atoms / List Item',
  component: ListItem,
  args,
} as Meta;

export const Normal: Story<typeof args> = (props) => <ListItem {...props} />;
