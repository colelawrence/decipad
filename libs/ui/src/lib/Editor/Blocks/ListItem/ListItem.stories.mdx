import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { ListItem } from './ListItem.component';
import { UnorderedList } from '@chakra-ui/layout';

<Meta
  title="Legacy/Editor/Blocks/Lists/ListItem"
  parameters={{
    docs: {
      source: {
        code: `<ListItem
  element={{
    children: [{ text: 'Hello World' }],
  }}
  attributes={{ 'data-slate-node': 'element', ref: null }}
>
  Hello World
</ListItem>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'Hello World',
      description: 'The text content of the component',
      type: 'string',
    },
  }}
  component={ListItem}
/>

# ListItem

The ListItem component is just a normal `li` element that has been adapted to be a slate specific block element.

export const Template = (args) => (
  <UnorderedList>
    <ListItem {...args} />
  </UnorderedList>
);

<Canvas>
  <Story name="ListItem">{Template.bind({})}</Story>
</Canvas>

<Props story="ListItem" />
