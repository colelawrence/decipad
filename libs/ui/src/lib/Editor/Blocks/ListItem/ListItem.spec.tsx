import { UnorderedList } from '@chakra-ui/layout';

import { render } from 'test-utils';
import { ListItem } from './ListItem.component';

describe('ListItem Block', () => {
  it('renders', () => {
    const { container } = render(
      <UnorderedList>
        <ListItem
          element={{
            children: [{ text: 'Hello World' }],
          }}
          attributes={{ 'data-slate-node': 'element', ref: null }}
        >
          Hello World
        </ListItem>
      </UnorderedList>
    );

    expect(container).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <UnorderedList>
        <ListItem
          element={{
            children: [{ text: 'Hello World' }],
          }}
          attributes={{ 'data-slate-node': 'element', ref: null }}
        >
          Hello World
        </ListItem>
      </UnorderedList>
    );

    expect(container).toMatchSnapshot();
  });
});
