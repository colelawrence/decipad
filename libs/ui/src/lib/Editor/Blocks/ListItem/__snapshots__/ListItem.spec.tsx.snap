// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Block matches snapshot 1`] = `
.emotion-0 {
  list-style-type: initial;
  -webkit-margin-start: 1em;
  margin-inline-start: 1em;
}

.emotion-1 {
  margin-left: 20px;
}

<div>
  <ul
    class="emotion-0"
    role="list"
  >
    <li
      class="emotion-1"
      data-slate-node="element"
    >
      Hello World
    </li>
  </ul>
</div>
`;
